cmd/iam.go

```diff
@@ -51,6 +54,40 @@ func getHash(text string) string {
 	return fmt.Sprintf("%x", h.Sum32())
 }
 
+// GetInstanceIamRole get instance iam role from metadata service
+func GetInstanceIamRole() (string, string, error) {
+	sess, err := session.NewSession()
+	if err != nil {
+		return "", "", err
+	}
+	metadata := ec2metadata.New(sess)
+	if !metadata.Available() {
+		return "", "", fmt.Errorf("EC2 Metadata is not available, are you running on EC2?")
+	}
+	resp, err := http.Get("http://169.254.169.254/latest/meta-data/iam/security-credentials/")
```

Comment:
Let me fix the http request thing, I wanted to be consistent with the current operation mode, same as the --default-role but auto discovered and set to the iam instance role.

The base arn is crucial, because it changes from account to account, we could have standard IAM role names but the base will change, keeping track of those base arn is a pain.

What I propose is to break this into two flags:
--auto-discover-base-arn
--auto-discover-default-role